presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -150,6 +152,28 @@ void testNoORWarning()
         assertNoWarning(analyzeWithWarnings("SELECT * FROM t1 JOIN t2 ON t1.a = t2.a \n" + "AND (t1.b = t2.b OR t1.b > t2.b)"));
     }
 
+    @Test
+    void testIgnoreNullWarning()
+    {
+        List<String> valueFunctions = ImmutableList.of(
+                "NTH_VALUE(c, 1)",
+                "FIRST_VALUE(c)",
+                "LAST_VALUE(c)",
+                "LEAD(c, 1)",
+                "LAG(c, 1)");
+
+        for (String function : valueFunctions) {
+            assertNoWarning(analyzeWithWarnings("SELECT a, " + function + " IGNORE NULLS OVER\n" +
+                    "(ORDER BY b)\n FROM (VALUES (1, 1, 3), (1, 2, null), (1, 4, 2)) AS t(a, b, c)"));
+        }
+
+        assertHasWarning(
```

Comment:
Please add more tests with various aggregate functions and those in this list https://prestodb.io/docs/current/functions/window.html#ranking-functions